### PR TITLE
Add ResolutionRequest v1beta1, without serving yet

### PR DESCRIPTION
**Pull request by abayer**
# Changes

part of #4710 

This is the first part of moving remote resolution to beta. Here, we add a new `v1beta1` version of `ResolutionRequest`, with conversion, though the new version is neither served nor stored. The difference between the `v1alpha1` and `v1beta1` version is that we're switching from `Parameters`, a `map[string]string`, to `Params`, a `[]pipelinev1beta1.Param`, so that we can end up passing array and object parameters to resolution requests.

/kind feature

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/resolution/v1beta1/resolution_request_conversion.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	}
+	return fmt.Errorf("v1beta1 is the highest known version, got: %T", sink)
+}
+
+// ConvertFrom implements apis.Convertible
+func (rr *ResolutionRequest) ConvertFrom(ctx context.Context, source apis.Convertible) error {
+	if apis.IsInDelete(ctx) {
+		return nil
+	}
+	return fmt.Errorf("v1beta is the highest known version, got: %T", source)
```
Comment (lbernick) [inline pkg/apis/resolution/v1beta1/resolution_request_conversion.go:42]:
```suggestion
	return fmt.Errorf("v1beta1 is the highest known version, got: %T", source)
```

**Review by lbernick — APPROVED** [review]

**Comment by vdemeester** [comment]
@abayer needs a small rebase 🙏🏼

**Comment by abayer** [comment]
@vdemeester done!

**Review by jerop — APPROVED** [review]
/lgtm

**Pull request merged**